src/MoBi.Presentation/Presenter/EditContainerPresenter.cs

```diff
@@ -119,13 +119,7 @@ public ContainerMode ConfirmAndSetContainerMode(ContainerMode newContainerMode)
       {
          var oldContainerMode = _container.Mode;
 
-         if (_isNewEntity)
-         {
-            _container.Mode = newContainerMode;
-            return newContainerMode;
-         }
-
-         if (newContainerMode == ContainerMode.Logical)
+         if (!_isNewEntity && newContainerMode == ContainerMode.Logical)
```

Comment:
it seems to me that with this modification: if I change a Container mode several times back and forth in the **New Container** dialog - multiple MacroCommands will be created at the end instead of just one.
Is it so?

Reply:
Yeah, that's right. That reminds me of another point -> in a PKSim module, if you run this way, then you will get a confirmation about converting to a non-pk module. I think that's not what we want.

I wanted to avoid duplicating the logic in command/non-command way. I will revise this a bit to make sure the prompts won't show up until you actually click `OK` on the modal.

Reply:
> if I change a Container mode several times back and forth in the

lol. PEBCAK